content/08-mongodb/09-mongodb-indexes.mdx

```diff
@@ -0,0 +1,260 @@
+
+Indexes are an incredibly useful way to optimize querying your database and are recommended. Knowing when to use them is important, so making sure that you are adding them on database fields that are often being queried will keep your reads efficient without negatively impacting database size and write efficiency.
+
+## How to create an index
+Now that we have an understanding of what indexes are and when to use them, we can get into the method of creating one.
+
+Once you have identified a query that could benefit from indexing, you use the `createIndex()` method of MongoDB. The basic syntax is as follows:
+
+```
+db.COLLECTION_NAME.createIndex( { "FIELD_NAME": 1 } )
+```
+
+`FIELD_NAME` is the name of the field that you want to create the index on, and `1` dictates an ascending order.
+
+An example use of the method would look something like:
+
+```
+db.mycoll.createIndex( { "country": 1 } )
+```
+
+You can also pass multiple fields in the `createIndex()` method by creating a comma seperated list like the following:
+
+```
+db.COLLECTION_NAME.createIndex( { "FIELD_NAME_1": 1, "FIELD_NAME_2": -1 } )
+```
+## How to show indexes
+Once you have started creating indexes, you may want to check what indexes insist on your database instances. In MongoDB, you can use the `getIndexes()` method to return descriptions of all the indexes in a collection.
+
+The basic syntax for seeing all of your collection's indexes is:
+
+```
+db.COLLECTION_NAME.getIndexes()
+``` 
+
+Using the previous example from when we created an index, the following displays the method and what it would return.
+
+```
+db.mycoll.getIndexes()
+```
+with a return of:
+
+```
+[
+   {
+      "v" : 2,
+      "key" : {
+         "country" : 1
+      },
+      "name" : "country"
+   }
+ ]
+```
+Index information includes the keys and options used to create the index.
+
+<PrismaOutlinks>
+
+If you're interested in using full text indexes with MongoDB, Prisma has the [`fullTextIndex` preview feature](https://www.prisma.io/docs/concepts/components/prisma-schema/indexes#full-text-indexes-mysql-and-mongodb) which allows you to easily migrate full text indexes into a Prisma schema for type safety and protection against validation errors.
+</PrismaOutlinks>  
+
+## How to understand index performance
+Now with the ability to create and check what indexes exist on your collection, you will want to see if your indexes are performing the way you expect.
+
+To begin the example, we will work with the `sample_mflix` database and the `comments` collection which has ~50.3k documents. This is a sample collecton provided by MongoDB simulating a datastore of movie and tv comments. 
```

Comment:
Can we link to this database?

Reply:
I can link to the MongoDB university where I got it from. It's their free education for MongoDB